pkg/kcp/provider/gcp/vpcpeering/loadKymaNetwork.go

```diff
@@ -31,6 +31,13 @@ func loadKymaNetwork(ctx context.Context, st composed.State) (error, context.Con
 		Name:      state.ObjAsVpcPeering().Spec.Details.LocalNetwork.Name,
 	}, network)
 
+	if composed.IsMarkedForDeletion(state.ObjAsVpcPeering()) &&
+		(client.IgnoreNotFound(err) != nil ||
+			apierrors.IsNotFound(err) ||
+			!meta.IsStatusConditionTrue(ptr.Deref(network.Conditions(), []metav1.Condition{}), cloudcontrolv1beta1.ConditionTypeReady)) {
+		return nil, nil
+	}
+
```

Comment:
If we want to allow our deletion process to continue when Kyma network is missing, perhaps it would be better to change the block below that that handles `if apierrors.IsNotFound(err) {`